code/datums/weather/weather_types/solar_flare.dm

```diff
@@ -35,9 +35,21 @@
 	// Solars produce 40x as much power. 240KW becomes 9.6MW. Enough to cause APCs to arc all over the station if >=2 solars are hotwired.
 	SSsun.solar_gen_rate = initial(SSsun.solar_gen_rate) * 40
 
+/datum/weather/solar_flare/can_weather_act(mob/living/L)
+	. = ..()
+	if(.) //If true the mob is already affected, no need to keep processing
+		return TRUE
+	if(!. && istype(L, /mob/living/simple_animal)) //while this might break immersion, I don't want to spam the server with calling this on simplemobs
```

Comment:
Don't need the `!.` since you return in the case it is true

```suggestion
	if(istype(L, /mob/living/simple_animal)) //while this might break immersion, I don't want to spam the server with calling this on simplemobs
```

Reply:
oh ye lmao i moved it up and forgot to yeet this, thanks